src/agents/sentiment.py

```diff
@@ -13,15 +17,15 @@ def sentiment_agent(state: AgentState):
     show_reasoning = state["metadata"]["show_reasoning"]
 
     # Loop through the insider trades, if transaction_shares is negative, then it is a sell, which is bearish, if positive, then it is a buy, which is bullish
-    signals = []
-    for trade in insider_trades:
-        transaction_shares = trade["transaction_shares"]
-        if not transaction_shares:
-            continue
-        if transaction_shares < 0:
-            signals.append("bearish")
-        else:
-            signals.append("bullish")
+
+    # dropping na values 
+    transaction_shares = pd.Series([t['trade'] for t in insider_trades]).dropna()
```

Comment:
Hey @PragyanTiwari - "trade" is not a valid key in the `t` here.

Perhaps you meant `"transaction_shares"`?